### PR TITLE
Cw adapt to v4 metadata 408

### PR DESCRIPTION
**Pull request by rexwangcc**
This PR is intended to embrace the v4(4.6.1 for now) metadata schema released in this quarter. 

This PR mainly related to the interaction between the Purple Box and Green Box, there is another PR about the connection between Blue Box and Green Box: https://github.com/HumanCellAtlas/lira/pull/45

I did the following things in this PR:

1. Update the **ss2/adapter.wdl**:
	- `sample_id = assay_json['sample_id']` -> `sample_id = assay_json['has_input']`
	- `fastq_1_name = assay_json['seq']['lanes'][0]['r1']` -> `fastq_1_name = assay_json['content']['seq']['lanes'][0]['r1']`
	- `fastq_2_name = assay_json['seq']['lanes'][0]['r2']` -> `fastq_2_name = assay_json['content']['seq']['lanes'][0]['r2']`
	- `from pipeline_tools import utils` -> `from pipeline_tools import dcp_utils`
	- `utils.get_manifest_files()` -> `dcp_utils.get_manifest_files()`
	- `utils.get_file_by_uuid()` -> `dcp_utils.get_file_by_uuid`
2. Update the **10x/adapter.wdl**:
	- `sample_id = assay_json['sample_id']` -> `sample_id = assay_json['has_input']`
	- `lanes = assay_json['seq']['lanes']` -> `lanes = assay_json['content']['seq']['lanes']`
	- `from pipeline_tools import utils` -> `from pipeline_tools import dcp_utils`
	- `utils.get_manifest_files()` -> `dcp_utils.get_manifest_files()`
	- `utils.get_file_by_uuid()` -> `dcp_utils.get_file_by_uuid`
3. Update the **submit.wdl** and the **pipeline-tools/create_analysis_json.py**:
	- add `create_core(type, schema_version)` to pipeline-tools/create_analysis_json.py for creating `core` object.
	- add unit tests
	- fix `setup.py`
4. Fix the **pipeline-tools/create_envelop.py** function:
	- add `schema_version = analysis_json['metadata_schema']` to get the schema version
	- add `'core': create_core(type='file', schema_version=schema_version)` in output_ref to match the new schema
5. Update the **docker image** `humancellatlas/pipeline-tools:0.1.4`:
	- add `.dockerignore` file
	- update the version of `hca-cli`
	- rebuild and push the docker: `humancellatlas/pipeline-tools:0.1.5`
6. Add examples:
	- add 10x/adapter_example_static_v4.json
	- add ss2/adapter_example_static_v4.json
7. Update adapter wdls to use the new docker: `humancellatlas/pipeline-tools:0.1.5`
8. Request changes to be made in [dcp int test](https://github.com/HumanCellAtlas/dcp/blob/4bc8deb37ee835ea649f996f863d1637897fcd03/tests/test_end_to_end_dcp.py#L106) to make sure the test pass for the schema change.

**Note**: Since the new schema requires schema_url in core object, in the future we may update the schema version in static input frequently.

### CODE REVIEW

File: adapter_pipelines/10x/adapter.wdl [hunk 1]
```diff
@@ -10,7 +10,7 @@ task GetInputs {
 
   command <<<
     python <<CODE
-    from pipeline_tools import utils
+    from pipeline_tools import dcp_utils
```
Comment (samanehsan) [inline adapter_pipelines/10x/adapter.wdl]: Do you also need to update the docker container so that it has the `dcp_utils` file?

Comment (rexwangcc) [reply adapter_pipelines/10x/adapter.wdl]: Yes, I did. Now it is `humancellatlas/pipeline-tools:0.1.5`.

Comment (samanehsan) [reply adapter_pipelines/10x/adapter.wdl]: Great! It should also be updated in this task too, which is currently still `humancellatlas/pipeline-tools:0.1.4`: https://github.com/HumanCellAtlas/pipeline-tools/blob/53220670659f173571519598745babfae793b21c/adapter_pipelines/10x/adapter.wdl#L186

Comment (rexwangcc) [reply adapter_pipelines/10x/adapter.wdl]: Thanks for this catch! I planned to update all the docker tags once I make the dcp-wide integration test works again. Also, I need to update some of the files according to the v4.6.0 metadata schema update.

File: tests/test_create_analysis_json.py [hunk 2]
```diff
@@ -70,8 +72,37 @@ def test_get_format(self):
...
+    def test_create_core_failure(self, mock):
+        type = 'analysis'
+        schema_version = 'bad_version'
+        schema_url = 'https://raw.githubusercontent.com/HumanCellAtlas/metadata-schema/{0}/json_schema/analysis.json'.format(schema_version)
+
+        mock.head(schema_url, status_code=404)
+
+        try:
+            core = analysis_json.create_core(type=type, schema_version=schema_version)
+        except HTTPError:
```
Comment (samanehsan) [inline tests/test_create_analysis_json.py]: Could we update the test to fail if an error is not raised? Right now it looks like it would pass either way

**Comment by samanehsan** [comment]
@rexwangcc just one comment about the tests otherwise this looks great!

**Comment by rexwangcc** [comment]
Pushed the latest docker image `humancellatlas/pipeline-tools:0.1.5` and `quay.io/humancellatlas/secondary-analysis-pipeline-tools:0.1.5`, and the DCP-wide integration test passed. Going to merge this PR.

**Pull request merged**